### PR TITLE
[Docs] Trans realization of distributed transaction into en

### LINKED ISSUE #6070: [Translation] Translate technical articles concerning Apache ShardingSphere
Hi, community,

This issue is to call for volunteers to **translate some technical articles from Chinese into English**. Anyone who enjoys translation or wants to make some contributions to open source community is warmly welcomed! 👏 

### Competencies needed
1. English translation skill
2. Basically, know what ShardingSphere is doing

### Article list
- [x] 1. [Start ElasticJob off with new goal](https://mp.weixin.qq.com/s/XJFj_vOj3bR6HRQUzy1ikg) --> @rudy2steiner
- [x] 2. [How to construct the distributed database](https://mp.weixin.qq.com/s/YSoX0Apq6vQeu8TU_53EAw) --> @EasonFeng5870
- [x] 3. [Apache ShardingSphere is included in CNCF Landscape](https://mp.weixin.qq.com/s/hAxON1CMYvgFf6sFIHtDHA) --> @mlw12345678
- [x] 4. [The mixed open-source distributed transaction solution](https://mp.weixin.qq.com/s/fYESyuw7O4u7wB-tSWlhyg) --> @mlw12345678
- [x] 5. [Would you like to become an Apache committer](https://mp.weixin.qq.com/s/sFqOz10KTk23IhmmDTy67g) --> @mlw12345678
- [x] 6. [How automatic executor of ShardingSphere works](https://mp.weixin.qq.com/s/hIKZPJDfGfeduW4frwVN_w)
- [x] 7. [How to merge the child resultsets](https://mp.weixin.qq.com/s/5__SGXEdvGqkVhpnrW6zug)
- [ ] 8. [The quick explanation of ShardingSphere transaction module](https://mp.weixin.qq.com/s/0LtqRTL8HewUAdZCk8LRLg)
- [x] 9. [What is the ShardingProxy](https://mp.weixin.qq.com/s/bTX0n9tH4x8eue19b7rdGw) --> @wcsoft 

### How to submit your translation
1. Add your translation to `xxx.en.md` in [blog](https://github.com/apache/shardingsphere/tree/master/docs/blog/content/material).

2. Raise a PR for review.

### Notice
1. All the Chinese articles named `xxx.cn.md` could be found in the same directory of Your translation as well. 
2. If you want o find a partner to work together for these long articles, please be free to give `recruit comment` below. 😉 
3. You can give a brief translation for these author profiles in the articles. Like just translate their names and titles.

If you want to have a try, **please comment here with** `Article number and your estimated deadline`. 

Besides, **any feedback and comments are welcomed as well**! 😃

### PR DESCRIPTION
**Pull request by rudy2steiner**
Fixes #6070.

Changes proposed in this pull request:
- Add  translation for  distributed transaction  of SS blog 
 
Corresponding Chinese commented content will remove once the translation ready to merge

### CODE REVIEW

File: docs/blog/content/material/realization.en.md [hunk 1]
```diff
@@ -4,4 +4,605 @@ weight = 12
...
+
+<!-- 2PC模型中，在prepare阶段需要等待所有参与子事务的反馈，因此可能造成数据库资源锁定时间过长，不适合并发高以及子事务生命周长较长的业务场景。 -->
+
+Sharding-Sphere supports XA-based strong consistent transaction solution, third-party XA implementations, such as Atomikos and Narayana, can be inject into Sharding-Sphere by SPI.
+<!-- Sharding-Sphere支持基于XA的强一致性事务解决方案，可以通过SPI注入不同的第三方组件作为事务管理器实现XA协议，如Atomikos和Narayana。 -->
+
+
+<!-- #### 柔性事务 -->
+
+### Soft Transaction
```
Comment (tristaZero) [inline docs/blog/content/material/realization.en.md:156]: Please use `BASE` Transaction.

File: docs/blog/content/material/realization.en.md [hunk 2]
```diff
@@ -4,4 +4,605 @@ weight = 12
...
+
+<!-- 柔性事务是对XA协议的妥协和补偿，它通过对强一致性要求的降低，以达到降低数据库资源锁定时间的效果。柔性事务的种类很多，可以通过各种不同的策略来权衡使用。 -->
+
+Soft transaction is a compromise to XA, don't need data to be strong consistent any more which benefit for reducing lock time of resource. There are a variety of soft transaction implementations, various strategies can be used to trade off.
```
Comment (tristaZero) [inline docs/blog/content/material/realization.en.md:160]: which` benefits` for?

File: docs/blog/content/material/realization.en.md [hunk 3]
```diff
@@ -4,4 +4,605 @@ weight = 12
...
+<!-- **一阶段提交 + 补偿 ：最大努力送达（BED）** -->
+**One Phase Commit with Compensation ：Best Effort Delivery(BED)**
+
+BED is a compensate strategy for weak XA. Using transaction table records all transactional operation SQL, the record will be deleted if sub transaction success to commit or retry until max configured retries. Trying it best to make data consistent that's why we call it BED. In different business scenes, in order to balance consistency and availability, using synchronous or asynchronous retry.
+
+<!-- 最大努力送达，是针对于弱XA的一种补偿策略。它采用事务表记录所有的事务操作SQL，如果子事务提交成功，将会删除事务日志；如果执行失败，则会按照配置的重试次数，尝试再次提交，即最大努力的进行提交，尽量保证数据的一致性，这里可以根据不同的业务场景，平衡C和A，采用同步重试或异步重试。
+
+这种策略的优点是无锁定资源时间，性能损耗小。缺点是尝试多次提交失败后，无法回滚，它仅适用于事务最终一定能够成功的业务场景。因此BED是通过事务回滚功能上的妥协，来换取性能的提升。 -->
+
+Advantage of the strategy is free of resource lock, low performance loss. But the transaction can't be rollback if still failed until reaches max reties. so it's only useful for the transaction that will be successful one hundred percentage. For the sake of improving performance, BED sacrifices rollback function of transaction.
```
Comment (tristaZero) [inline docs/blog/content/material/realization.en.md:172]: `The` Advantage of the strategy?

File: docs/blog/content/material/realization.en.md [hunk 4]
```diff
@@ -4,4 +4,605 @@ weight = 12
...
+
+![](https://shardingsphere.apache.org/blog/img/realization4.jpg)
+
+<!-- **TCC： Try-Confirm-Cancel**   -->
+
+**TCC： Try-Confirm-Cancel**  
+
+<!-- TCC模型是把锁的粒度完全交给业务处理，它需要每个子事务业务都实现Try-Confirm/Cancel接口。 -->
+TCC model gives the control of lock to business level, and all sub transactions business must be implement Try-Confirm/Cancel interface.
```
Comment (tristaZero) [inline docs/blog/content/material/realization.en.md:181]: be `implemented`?

File: docs/blog/content/material/realization.en.md [hunk 5]
```diff
@@ -4,4 +4,605 @@ weight = 12
...
+
+    Cancel execute business;
+
+    Release reserved resource in the Try phase;
+
+    Cancel operation is idempotent;
+
+<!-- 这三个阶段都会按本地事务的方式执行，不同于XA的prepare，TCC无需将XA的投票期间的所有资源挂起，因此极大的提高了吞吐量。   -->
+
+Different from XA prepare stage, the three stage will execute as the local transaction way. TCC don't need to lock all the resource during voting, and improves system concurrency.
```
Comment (tristaZero) [inline docs/blog/content/material/realization.en.md:238]: the three stage?

File: docs/blog/content/material/realization.en.md [hunk 6]
```diff
@@ -4,4 +4,605 @@ weight = 12
...
+    检查A账户余额是否充足；
+
+    从A账户中扣减100元，并将状态置为“转账中”；
+
+    预留扣减资源，将从A往B账户转账100元这个事件存入消息或者日志中； -->
+*   **Try：**
+
+    Check effectiveness of account A, namely, check status of account A whether in "transferring" or "Frozen";
+
+    Check account A whether has enough money;
```
Comment (tristaZero) [inline docs/blog/content/material/realization.en.md:268]: `whether` account A has ?

File: docs/blog/content/material/realization.en.md [hunk 7]
```diff
@@ -4,4 +4,605 @@ weight = 12
...
+
+<!-- **消息驱动** -->
+
+**Message-Driven Transaction Model**
+
+![](https://shardingsphere.apache.org/blog/img/realization6.jpg)
+
+<!-- 消息一致性方案是通过消息中间件保证上下游应用数据操作的一致性。基本思路是将本地操作和发送消息放在一个事务中，下游应用向消息系统订阅该消息，收到消息后执行相应操作。本质上是依靠消息的重试机制，达到最终一致性。消息驱动的缺点是：耦合度高，需要在业务系统中引入MQ，导致系统复杂度增加。 -->
+
+Message based consistent solution depends on message middleware to make sure upstream and downstream applications keep the data consistent. The basic idea is put the local operation and sending message into a local transaction, downstream application consumes the message and execute the corresponding operation. It essentially relies on retry mechanism of message middleware to achieve eventually consistent . The disadvantage of message-driven is highly couple with message queue, which may increasing the complexity of business system.
```
Comment (tristaZero) [inline docs/blog/content/material/realization.en.md:339]: The basic idea `is put `?

Comment (tristaZero) [reply docs/blog/content/material/realization.en.md:339]: which may` increase`?

File: docs/blog/content/material/realization.en.md [hunk 8]
```diff
@@ -4,4 +4,605 @@ weight = 12
...
+Either the sequence T1, T2, ..., Tn(which is the preferable one) or T1, T2, ...,Tx, Cx, ..., C2, C1 will be executed.
+
+<!-- 当每个Saga子事务 T1, T2, …, Tn 都有对应的补偿定义 C1, C2, …, Cn-1,那么Saga系统可以保证：
+
+*   子事务序列 T1, T2, …, Tn得以完成 (最佳情况)；
+
+*   或者序列 T1, T2, …, Tj, Cj, …, C2, C1, 0 < j < n, 得以完成。 -->
+
+
+ For the lack of prepare stage on Saga model, transactions can't keep isolation to each other. So loss of update, dirty read and et problem will happen when a resource is operated concurrently by multi transactions. The problem can be settle by concurrency control in business level, such as lock or pre-allocation resource.
```
Comment (tristaZero) [inline docs/blog/content/material/realization.en.md:367]: can be `settled`?

File: docs/blog/content/material/realization.en.md [hunk 9]
```diff
@@ -4,4 +4,605 @@ weight = 12
...
+
+All in word, TCC and MQ are both depend on business transformation, but XA, BED and SAGA only involve database. So latter is more preferable solution, which is less invasive for business, and low transformation cost.
+
+
+<!-- ### Sharding-Sphere对分布式事务的支持 -->
+### Distributed Transaction of Sharding-Sphere
+
+<!-- Sharding-Sphere是一套开源的分布式数据库中间件解决方案组成的生态圈，它由Sharding-JDBC、Sharding-Proxy和Sharding-Sidecar这3款相互独立的产品组成。它们均提供标准化的数据分片、读写分离、柔性事务和数据治理功能，可适用于如Java同构、异构语言、容器、云原生等各种多样化的应用场景。 -->
+
+[ShardingSphere](https://github.com/sharding-sphere/sharding-sphere/) is an open-source ecosystem consisted of a set of distributed database middleware solutions, including 3 independent products, JDBC, Proxy & Sidecar. They all provide functions of data sharding, distributed transaction and database governance, applicable in a variety of situations such as Java isomorphism, heterogeneous language container and and cloud native.
```
Comment (tristaZero) [inline docs/blog/content/material/realization.en.md:405]: `and and` cloud native?

File: docs/blog/content/material/realization.en.md [hunk 10]
```diff
@@ -4,4 +4,605 @@ weight = 12
...
+As can be seen from above figure, Sharding-core will produce and dispatch various events according SQL, and the thread of transaction listener will invoke corresponding transaction processor to handle when event arrives.
+
+<!-- #### Sharding-Proxy事务实现 -->
+
+### Transaction Implementation of Sharding-Sphere
+
+
+<!-- Sharding-Proxy是基于netty开发的数据库中间代理层，实现了标准的MySQL协议，可以看做是一个实现了数据分片的数据库。Sharding-Proxy已经实现了基于Atomikos的XA事务，为了保证所有的子事务都处于同一个线程之中，整个Proxy的线程模型进行了如下的调整： -->
+
+Sharding-Proxy is a netty based database middle layer proxy, it implements MySQL protocol, and could be regard as a database with built-in data sharding ability. Sharding-Proxy have implemented XA transaction based on Atomikos. For ensuring all sub transaction in the same thread, the changes of proxy thread model can be seen from the following figure.
```
Comment (tristaZero) [inline docs/blog/content/material/realization.en.md:445]: could be `regarded`?

File: docs/blog/content/material/realization.en.md [hunk 11]
```diff
@@ -4,4 +4,605 @@ weight = 12
...
+
+![](https://shardingsphere.apache.org/blog/img/realization9.jpg)
+
+<!-- 当开启事务后，Proxy后端的SQL命令执行引擎将采用一通道一线程的模式，此事务线程的生命周期同通道保持一致。事务处理的具体过程与Proxy彻底解耦，即Proxy将发布事务类型的事件，然后Sharding-Sphere-TM根据传入的事务消息，选择具体的TM进行处理。 -->
+
+When transaction enabled, SQL engine will using Channel-Thread pattern on Proxy backend, and lifecycle of channel and transaction thread keeps same. The process procedure of transaction are totally decoupled with Proxy, namely, the transaction event produced by Proxy will consumed by Sharding-Sphere TM.
```
Comment (tristaZero) [inline docs/blog/content/material/realization.en.md:451]: SQL engine `will use`?

File: docs/blog/content/material/realization.en.md [hunk 12]
```diff
@@ -4,4 +4,605 @@ weight = 12
...
+Imbeded Saga Transaction Manager of Sharding-Sphere
+
+![](https://shardingsphere.apache.org/blog/img/realization13.jpg)
+
+<!-- Saga以jar包的形式提供分布式事务治理能力。 -->
+Saga provides distributed transaction service governance in form of jar.
+
+<!-- 对Sharding-Sphere而言，confirm和cancel过程代表了子事务中的正常执行SQL和逆向执行SQL，（未来Sharding-Sphere将提供自动生成逆向SQL的能力）。当启用Saga柔性事务后，路由完成之后的物理数据源将开启本地自动提交事务，每次confirm和cancel都会直接提交。 -->
+
+For Sharding-Sphere, procedure of confirm and cancel represent normal and backward execution SQL of sub-transaction. Automation of reverse
```
Comment (tristaZero) [inline docs/blog/content/material/realization.en.md:512]: `procedure of confirm `?

File: docs/blog/content/material/realization.en.md [hunk 13]
```diff
@@ -4,4 +4,605 @@ weight = 12
...
+
+<!-- **Q5**：如果我只想用SS里的事务模块，可以吗？ -->
+**Q5**: If I only want to use transaction module of SS, is it possible?
+
+<!-- **A5**：SS是以事件驱动的方式进行的架构，未来事务模块只负责事务相关的处理。 -->
+**A5**: Architecture of SS is event-driven, transaction module will only responsible for transaction in the future.
+
+
+<!-- **Q6**：SAGA不支持ACID中的I，咱们这边怎么考虑的呢？ -->
+**Q6**: SAGA don't support I of ACID, how do you think of this ?
```
Comment (tristaZero) [inline docs/blog/content/material/realization.en.md:594]: SAGA don't support I of ?

File: docs/blog/content/material/realization.en.md [hunk 14]
```diff
@@ -4,4 +4,605 @@ weight = 12
...
+<!-- **A6**：目前暂不支持隔离性，今后我们有增加I的规划，其实所有的柔性事务都不支持I，TCC增加了Try阶段，可以理解是准隔离性，使用SAGA时，可以在业务层面控制并发，防止脏读等产生。 -->
+**A6**: Now, Isolation is unsupported, and we have plan to support it. Actually all of soft transaction don't support I, Try stage of TCC can be regarded as isolation. Control of concurrency on business level can avoid dirty read.
+
+
+<!-- **Q7**：那意思，现在3的版本还不能单独用事务的模块？ -->
+**Q7**: In version 3.0, does module of transaction works independently ?
+
+<!-- **A7**：现在3.0版本，事务模块依赖了Sharding-JDBC模块，事务模块需要监听Sharding-JDBC和Proxy中的事件，然后进行事务操作。如果你想单独用事务模块，需要按Core中定义的事件，在你的业务里进行发布。 -->
+
+ **A7**: In version 3.0, transaction module depends on Sharding-JDBC module, and executes transaction when received the event of Sharding-JDBC and Proxy. If you want to use transaction module, you need to handle event according the definition on core module.
```
Comment (tristaZero) [inline docs/blog/content/material/realization.en.md:604]: according ?

**Review by tristaZero — CHANGES_REQUESTED** [review]
I gave a rough review of this translation. Basically, it mostly gave an accurate translation. However, It still needs more grammar and word checking. I guess some of the grammar-checking tools or APPs can help. Please give a look at my comments and very appreciated. : )

**Comment by rudy2steiner** [comment]
Thanks for your review and suggestions, And I have checked and updated

**Review by tristaZero — APPROVED** [review]
Hi @rudy2steiner Thanks for your time and efforts on this long article. : )

I suppose this translation will be polished by another professional translator later.

BTW, Merry Christmas early!

**Comment by rudy2steiner** [comment]
OK, Merry Christmas!

**Pull request merged**